### DISCUSSION TITLE
[Help]: Auto Height doesn't work when --slide-size: 50%

### DISCUSSION THREAD

**Discussion by Enkratia** [davidjerleke/embla-carousel#1053] [Help] [answered]
### Summary

Embla respects only first slide height, second slide is partially hidden (as it is higher than the first one).
![embla_auto_height](https://github.com/user-attachments/assets/6bd368cd-411f-4a10-8a84-695aa9962578)

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [X] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Additional information

```js
Is this expected behaviour?
```

### CodeSandbox example
https://codesandbox.io/p/sandbox/embla-carousel-auto-height-react-forked-x4qk9h?file=%2Fpackage.json&workspaceId=90329704-cfa2-4724-99d1-73fb71a905c4

**Comment by davidjerleke** [comment]
@Enkratia thanks for your question. Could you give me access to the sandbox please? You need to make it public.

**Reply by Enkratia** [reply]
@davidjerleke, `CodeCandbox` link is public now.

** this is the same sandbox as https://codesandbox.io/p/sandbox/68hflr (docs)
** one single change:  `--slide-size: 50%;` instead of   `--slide-size: 100%;`

**Reply by davidjerleke** [reply, marked as answer]
@Enkratia thanks. Yes this is expected. Embla will get the highest slide for the selected snap index. You can add:

```tsx
const options = { slidesToScroll: 'auto' }
```

And group your slides to solve this.

**Reply by Enkratia** [reply]
@davidjerleke, Thank you
